packages/ui-markdown-editor/src/lib/utilities/toolbarHelpers.js

```diff
@@ -40,11 +40,29 @@ export const toggleBlock = (editor, format) => {
 
     if (isList(format)) {
       const listItemBlock = { type: LIST_ITEM, children: [], data: { tight: true } };
-      // eslint-disable-next-line no-restricted-syntax
+      let anchor = [];
+      let focus = [];
+      editor.selection.anchor.path.map((item,index)=>{
+      if(index===editor.selection.anchor.path.length-1){
+        anchor[index]=0
+        anchor[index+1]=item
+      }else{
+        anchor[index]=item
+      }
+       });
+      editor.selection.focus.path.map((item,index)=>{
+        if(index===editor.selection.focus.path.length-1){
+          focus[index]=0
+          focus[index+1]=item
+        }else{
+          focus[index]=item
+        }
+        });
+      // eslint-disable-next-line no-restricted-syntax  
       for (const [node, path] of Node.descendants(
         editor,
-        { from: editor.selection.anchor.path, to: editor.selection.focus.path }
-      )) {
+        { from: anchor, to: focus }
```

Comment:
Very nice solution! A thought on the implementation: Instead of mapping through the `anchor` and `focus`, could we do something like this?

```js
{
  from: editor.selection.anchor.path.splice((editor.selection.anchor.path.length - 1), 0, 0), 
  to: editor.selection.focus.path.splice((editor.selection.focus.path.length - 1), 0, 0)
}
```

Reply:
On second thought, it would need to be a method that returns a copy of the path array instead of mutating it.

Reply:
maybe `slice` would be an option here?

Reply:
Yes, `slice` and `concat` could work.

```javascript
a.slice(0, -1).concat(0, 0, a[a.length - 1])
```

Reply:
@KartikSoneji yes! do we need to add two zeroes to the array or just one?

Reply:
@DianaLease Sorry, I misread.
It should be:
```javascript
a.slice(0, -1).concat(0, a[a.length - 1])
```

Reply:
@KartikSoneji That looks good to me! @hohooio823 are you able to implement this approach?

Reply:
Looks good ! I'll implement them .